cargo-dist/src/lib.rs

```diff
@@ -72,9 +72,18 @@ pub fn do_build(cfg: &Config) -> Result<DistManifest> {
     }
     eprintln!();
 
-    // Run all the build steps
-    for step in &dist.build_steps {
-        run_build_step(&dist, step)?;
+    // Run all the local build steps first
+    for step in &dist.local_build_steps {
+        run_build_step(&dist, step, &[])?;
+    }
+
+    // Calculate a temporary build manifest from the output of the local builds
+    // (includes their linakge data)
+    let manifests = vec![build_manifest(cfg, &dist)?];
```

Comment:
Wow if this just works that's incredibly fucking elegant, kudos.

(It is *slightly* a lie because it claims the global things are already built, I think, but that's fine if global things don't depend on other global things. And if they did we'd need to do even more complex work beyond the scope of this PR.)

Reply:
AFAICT it's actually working! It's the same dist manifest we get via the "copy JSON back and forth between files" method, functionally, and linkage data is intact. Right now, global builds only depend on local builds, so I think this is safe.